plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/segmented/RocksDBSnapshot.java

```diff
@@ -47,13 +47,4 @@ void unMarkSnapshot() {
       dbSnapshot.close();
     }
   }
-
-  // TODO:  this is a stopgap measure, revisit with https://github.com/hyperledger/besu/issues/4641
-  @Override
-  protected void finalize() {
-    if (usages.decrementAndGet() > 0) {
```

Comment:
where are we closing now ?

Reply:
All of the uses of `getMutable` and  copy` are currently wrapped in try-with-resources.  In #4641 I will be adding static analysis via @MustBeClosed so we can be sure all uses of getMutable are safeguarded